### DISCUSSION TITLE
Compile syslog-ng from source on Mac M1 fails with unexpected token newline

### DISCUSSION THREAD

**Discussion by Genfood** [syslog-ng/syslog-ng#4342] [Q&A] [answered]
Hey folks,

I am currently trying to build syslog from source on a M1 Mac, but it fails if I try to run the configure script. I followed the available instructions, on how to build syslog-ng on Mac.

The CLI tells me:

```
% ./configure --with-ivykis=system --disable-java --prefix=$PWD/build
checking whether make supports nested variables... yes
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... ./install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking whether UID '501' is supported by ustar format... yes
checking whether GID '20' is supported by ustar format... yes
checking how to create a ustar tar archive... gnutar
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of gcc... gcc3
./configure: line 6168: syntax error near unexpected token `newline'
./configure: line 6168: `    '''

```
I tried it with Apples clang compiler and I also installed a gcc version via brew. But I always got the same result. :( 
Here is the [log file](https://github.com/syslog-ng/syslog-ng/files/10824594/config.log):

And that's the error:
> gcc: error: unrecognized command-line option '-V'
> gcc: fatal error: no input files

Maybe someone knows what I am doing wrong here?

**Comment by czanik** [comment, marked as answer]
I do not have an ARM64 mac, just an old Intel MacBook Pro. I followed the instructions from https://www.syslog-ng.com/community/b/blog/posts/syslog-ng-on-macos-ventura and was able to build both syslog-ng 4.0.1 and the latest git snapshot.

For 4.0.1 I started out from the release tgz: wget https://github.com/syslog-ng/syslog-ng/releases/download/syslog-ng-4.0.1/syslog-ng-4.0.1.tar.gz

For the git snapshot is used syslog-ng dbld to build a release tgz on Linux and then copied the tgz to my aging Mac. From the configure output I guess you did the same (it mentions 4.0.1.222.dirty as version, which is a release name dbld gives to locally modified git snapshots).

On the Mac I started bash (yes, I'm too much a Linux user :-) ), extracted the source and then:

```
brew install automake autoconf binutils glib autoconf-archive flex bison libtool pkg-config ivykis openssl pcre json-c
brew install net-snmp
export PATH=/usr/local/opt/bison/bin:$PATH
export PKG_CONFIG_PATH="/usr/local/opt/openssl@1.1/lib/pkgconfig:$PKG_CONFIG_PATH"
export PATH="/usr/local/opt/net-snmp/bin:$PATH"
./configure --disable-java --with-ivykis=system --prefix=/Users/czanik/syslog-ng-4.0.1/sng
make
make install
```

Then I repeated the process for the git snapshot. The only difference was the configure --prefix argument:
```
./configure --disable-java --with-ivykis=system --prefix=/Users/czanik/syslog-ng-4.0.1.224.gf3974e3/sng
```

**note**: directory names are different on the ARM macs

In the end:
```
bash-3.2$ pwd
/Users/czanik/syslog-ng-4.0.1.224.gf3974e3/sng/sbin
bash-3.2$ ./syslog-ng -V
syslog-ng 4 (4.0.1.224.gf3974e3)
Config version: 4.0
Installer-Version: 4.0.1.224.gf3974e3
Revision: 
Compile-Date: Feb 24 2023 16:02:54
Module-Directory: /Users/czanik/syslog-ng-4.0.1.224.gf3974e3/sng/lib/syslog-ng
Module-Path: /Users/czanik/syslog-ng-4.0.1.224.gf3974e3/sng/lib/syslog-ng
Include-Path: /Users/czanik/syslog-ng-4.0.1.224.gf3974e3/sng/share/syslog-ng/include
Available-Modules: timestamp,kvformat,redis,riemann,appmodel,afprog,examples,rate-limit-filter,cef,map-value-pairs,afsnmp,kafka,stardate,system-source,confgen,afuser,xml,disk-buffer,tfgetent,linux-kmsg-format,correlation,json-plugin,add-contextual-data,pseudofile,affile,csvparser,basicfuncs,syslogformat,hook-commands,afmongodb,graphite,tags-parser,geoip2-plugin,afstomp,http,secure-logging,afsql,mod-python,afsocket,cryptofuncs,azure-auth-header,regexp-parser
Enable-Debug: off
Enable-GProf: off
Enable-Memtrace: off
Enable-IPv6: on
Enable-Spoof-Source: on
Enable-TCP-Wrapper: off
Enable-Linux-Caps: off
Enable-Systemd: off
bash-3.2$ 
```

**Comment by MrAnno** [comment]
Have you installed the Xcode command line tools? `xcode-select --install`
Instead of using Homebrew's gcc `CC='gcc'`, I'd try using the builtin clang compiler.

**Reply by MrAnno** [reply]
Here is the full guide with the necessary build dependencies:
https://syslog-ng.gitbooks.io/getting-started/content/chapters/chapter_0/section_3.html

Also:
https://github.com/Homebrew/homebrew-core/blob/955021b2c365f4d020968962f61f5f10aea1b120/Formula/syslog-ng.rb

**Reply by Genfood** [reply]
I tried both first with clang than with gcc, but that was just a test. But I switched now back to clang.

**Comment by czanik** [comment]
Yeah. As you can see from the list of packages I installed, I have a full Xcode installation and did not install compiler tools from Homebrew.

**Comment by Genfood** [comment]
I got it working, after a bit of fiddling around. But I don't know exactly what I did wrong, but it is working!

Steps I did:
- after a failed run, getting a clean repo again (I guess that was the main thing)
- Wasn't sure about if I have gmake and gm4 installed, but I had
- ...
- 
Thx for your help.

Edit:
I just tried it again and can confirm that the problem was resulting from not including the openssl@1.1 path to the `PKG_CONFIG_PATH` variable, even if the `PKG_CONFIG_PATH` has been updated. After a cleaning of the repository, it worked.

**Comment by julia-grdt** [comment]
Hi, I am currently having the exact same issue. I am on ubuntu 24.04 but on a Mac with an Intel chip. 

So when I do ./autogen.sh I get a lot of warnings, this is a snippet:

Makefile.am:183: library has 'NODIST_BUILT' as canonical name (possible typo)
configure.ac:464: warning: AC_PROG_CC_C99 is obsolete; use AC_PROG_CC
configure.ac:467: warning: ac_ext=c
configure.ac:467: ac_cpp='$CPP $CPPFLAGS'
configure.ac:467: ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
configure.ac:467: ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
configure.ac:467: ac_compiler_gnu=$ac_cv_c_compiler_gnu
configure.ac:467: if test -n "$ac_tool_prefix"; then
configure.ac:467:   # Extract the first word of "${ac_tool_prefix}gcc", so it can be a program name with args.
configure.ac:467: set dummy ${ac_tool_prefix}gcc; ac_word=$2
configure.ac:467: { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
configure.ac:467: printf %s "checking for $ac_word... " >&6; }
configure.ac:467: if test ${ac_cv_prog_CC+y}
configure.ac:467: then :
configure.ac:467:   printf %s "(cached) " >&6
configure.ac:467: else $as_nop
configure.ac:467:   if test -n "$CC"; then
configure.ac:467:   ac_cv_prog_CC="$CC" # Let the user override the test.
configure.ac:467: else
configure.ac:467: as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
**configure.ac:467: for as_dir in $PATH
configure.ac:467: do
configure.ac:467:   IFS=$as_save_IFS
configure.ac:467:   case $as_dir in #(((
configure.ac:467:     '' is m4_require'd but not m4_defun'd**
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/programs.m4:41: _AC_CHECK_PROG is expanded from...
./lib/autoconf/programs.m4:101: AC_CHECK_PROG is expanded from...
./lib/autoconf/programs.m4:221: AC_CHECK_TOOL is expanded from...
./lib/autoconf/c.m4:452: AC_PROG_CC is expanded from...


The bold part is also the line where the "newline" problem shows up when doing ./configure 

checking whether make supports the include directive... yes (GNU style)
checking dependency style of gcc... gcc3
./configure: line 6792: syntax error near unexpected token `newline'
./configure: line 6792: `    '''


At line 6792 the code is:

do
  IFS=$as_save_IFS
  case $as_dir in #(((
    ''
 as_dir=./ ;;
    */) ;;
    *) as_dir=$as_dir/ ;;
  esac
    for ac_exec_ext in '' $ac_executable_extensions; do
  if as_fn_executable_p "$as_dir$ac_word$ac_exec_ext"; then
    ac_cv_prog_CC="${ac_tool_prefix}gcc"
    printf "%s\n" "$as_me:${as_lineno-$LINENO}: found $as_dir$ac_word$ac_exec_ext" >&5
    break 2
  fi
  
  
  
 How can I solve this problem?

**Reply by HofiOne** [reply]
Hi,
Please
- use a clean repo
- follow the exact steps described [here](https://syslog-ng.github.io/dev-guide/chapter_4/section_2/README)

you will need all the dependencies enumerated on the page to get it to work (depending on the distro and its version, there might be further dependencies needed)

if you follow the description it must work
